sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -191,6 +218,67 @@ class MapboxMap internal constructor(
     styleExtension: StyleContract.StyleExtension
   ) = loadStyle(styleExtension, null, null)
 
+  /**
+   * Overrides the map style's transition options with user-provided options.
+   *
+   * The style transition is re-evaluated when a new style is loaded.
+   *
+   * Style transition options will be applied to current style immediately if it is loaded and
+   * to the upcoming style as well if [forUpcomingStyle] is True.
+   * If no style is loaded [transitionOptions] will be applied to an upcoming style not taking
+   * [forUpcomingStyle] flag in consideration.
+   */
+  @JvmOverloads
+  fun setStyleTransition(
+    transitionOptions: TransitionOptions,
+    forUpcomingStyle: Boolean = true
```

Comment:
shall we keep the `setStyleTransition` for both `MapboxMap` and `Style`? Then we can remove the flag, and only update the cached transition for `MapboxMap`. The reason is the settings in the `MapboxMap` should be independent of the `Style`, adding this flag could be confusing.

Reply:
If I'm the user, I expect the `setStyleTransition` in MapboxMap object to take effect immediately(pass through to gl-native) and also be cached for future transitions. Also I would suggest rename this method to 
```
fun setTransition(transitionOptions: TransitionOptions)
```

Reply:
refactored again